src/core/lib/surface/completion_queue.c

```diff
@@ -576,6 +549,42 @@ static void cq_check_tag(grpc_completion_queue *cq, void *tag, bool lock_cq) {
 static void cq_check_tag(grpc_completion_queue *cq, void *tag, bool lock_cq) {}
 #endif
 
+static bool cq_begin_op_for_next(grpc_completion_queue *cq, void *tag) {
+  cq_next_data *cqd = DATA_FROM_CQ(cq);
+  while (true) {
+    gpr_atm count = gpr_atm_no_barrier_load(&cqd->pending_events);
+    if (count == 0) {
+      cq_check_tag(cq, tag, true); /* Used in debug builds only */
```

Comment:
ok. It was not failing because just before calling this function, we were adding the tag to `cq->outstanding_tags` (in function grpc_cq_begin_op). That is not really the correct way to do it (because we may still have race conditions) but since its only for debugging purposes, I guess its ok.. In anycase, we should remove the `cq_check_tag` here.